discord/activity.py

```diff
@@ -825,7 +828,109 @@ def __repr__(self) -> str:
         return f'<CustomActivity name={self.name!r} emoji={self.emoji!r}>'
 
 
-ActivityTypes = Union[Activity, Game, CustomActivity, Streaming, Spotify]
+class HangStatus(BaseActivity):
+    """A slimmed down version of :class:`Activity` that represents a Discord hang status.
+
+    This is typically displayed via **Right now, I'm -** on the official Discord client.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two hang statuses are equal.
+
+        .. describe:: x != y
+
+            Checks if two hang statuses are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the hang status' hash.
+
+        .. describe:: str(x)
+
+            Returns the hang status' name.
+
+    .. versionadded: 2.5
+
+    Attributes
+    -----------
+    state: :class:`HangStatusType`
+        The state of hang status.
+    name: :class:`str`
+        The name of the hang status.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji of the hang status if any.
+    """
+
+    __slots__ = ('state', 'name', 'emoji')
+
+    def __init__(self, state: str, emoji: Optional[Union[PartialEmoji, Dict[str, Any], str]] = None, **extra: Any) -> None:
+        super().__init__(**extra)
+        self.state: HangStatusType = try_enum(HangStatusType, state)
+
+        self.name: str
+        if self.state == HangStatusType.custom:
+            self.name = extra['details']
+        else:
+            self.name = self.state.value
+
+        self.emoji: Optional[PartialEmoji]
+        if emoji is None:
+            self.emoji = emoji
+        elif isinstance(emoji, dict):
+            self.emoji = PartialEmoji.from_dict(emoji)
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji(name=emoji)
+        elif isinstance(emoji, PartialEmoji):
+            self.emoji = emoji
+        else:
+            raise TypeError(f'Expected str, PartialEmoji, or None, received {type(emoji)!r} instead.')
+
+    @property
+    def type(self) -> ActivityType:
+        """:class:`ActivityType`: Returns the activity's type. This is for compatibility with :class:`Activity`.
+
+        It always returns :attr:`ActivityType.hang`.
+        """
+        return ActivityType.hang
+
+    def to_dict(self) -> Dict[str, Any]:
+        if self.state == HangStatusType.custom:
+            ret = {
+                'type': ActivityType.hang.value,
+                'name': 'Hang Status',
+                'state': self.state.value,
+                'details': self.name,
+            }
+        else:
+            ret = {
+                'type': ActivityType.hang.value,
+                'name': 'Hang Status',
+                'state': self.state.value,
+            }
```

Comment:
```suggestion
        ret = {
            'type': ActivityType.hang.value,
            'name': 'Hang Status',
            'state': self.state.value,
        }

        if self.state == HangStatusType.custom:
            ret['details'] = self.name
```

Reply:
Yeah, I made it not that readable. Anyway, I removed it because bots can't use hang statuses anyway.